### PR TITLE
Added requirements and normalization

### PR DESCRIPTION
**Pull request by EleniSmyrniou**

### CODE REVIEW

File: CPTSuperLearn/utils_SchemaGAN.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    # Formula used for normalization is:
+    # normalized_data = 2 * (original_data / data_range) - 1
+    src_normalized = 2 * (data / data_range) - 1
+
+    return src_normalized
+
+
+
+
+def reverse_IC_normalization(data):
```
Comment (brunozc) [inline CPTSuperLearn/utils_SchemaGAN.py]:
```suggestion
def reverse_IC_normalization(data: np.ndarray):
```

File: CPTSuperLearn/utils_SchemaGAN.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import pandas as pd
+
+
+def IC_normalization(data):
```
Comment (brunozc) [inline CPTSuperLearn/utils_SchemaGAN.py]:
```suggestion
def IC_normalization(data: np.ndarray):
```

File: CPTSuperLearn/utils_SchemaGAN.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    data (list): List containing the source and target data.
+
+    Returns:
+    list: A list containing the normalized source and target data.
+    """
+
+    # Define the maximum and minimum values of IC in the source and target images
+    max_IC_value = 4.5  # Maximum expected IC value
+    min_IC_value = 0  # Minimum expected IC value, it's not really zero,
+    # but when deleting data it will be zero
```
Comment (brunozc) [inline CPTSuperLearn/utils_SchemaGAN.py]: maybe pass set these values as globals? they are used in both functions?

File: CPTSuperLearn/utils_SchemaGAN.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    data (np.array): Array containing the normalized data.
+
+    Returns:
+    np.array: An array containing the rescaled data.
+    """
+
+    # Define the maximum and minimum values of IC in the source and target images
+    max_IC_value = 4.5  # Maximum expected IC value
+    min_IC_value = 0  # Minimum expected IC value, it's not really zero,
+    # but when deleting data it will be zero
```
Comment (brunozc) [inline CPTSuperLearn/utils_SchemaGAN.py]: maybe pass set these values as globals? they are used in both functions?

File: CPTSuperLearn/utils_SchemaGAN.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    # Rescale the data to the original range [min_IC_value, max_IC_value]
+    # Formula used for rescaling is:
+    # rescaled_data = (normalized_data + 1) * (data_range / 2) + min_IC_value
+    X = (data + 1) * (data_range / 2) + min_IC_value
+
+    return X
+
+
+def preprocess_data(data: pd.DataFrame, columns_to_keep_index: list) -> np.ndarray:
```
Comment (brunozc) [inline CPTSuperLearn/utils_SchemaGAN.py]: missing docstring

File: requirements.txt [hunk 6]
```diff
@@ -4,4 +4,6 @@ scipy==1.14.1
 torch==2.5.1
 torchaudio==2.5.1
 torchvision==0.20.1
-tqdm==4.67.0
\ No newline at end of file
+tqdm==4.67.0
```
Comment (brunozc) [inline requirements.txt:7]: missing pandas?

File: setup.cfg [hunk 7]
```diff
@@ -22,6 +22,8 @@ install_requires =
     torchaudio==2.5.1
     torchvision==0.20.1
     tqdm==4.67.0
+    tensorflow==2.8.0
+    keras==2.8.0
```
Comment (brunozc) [inline setup.cfg:26]: missing pandas?

File: CPTSuperLearn/interpolator.py [hunk 8]
```diff
@@ -3,7 +3,7 @@
 import numpy as np
 from scipy.spatial import cKDTree
 from keras.models import load_model
-from utils import IC_normalization, reverse_IC_normalization
+from CPTSuperLearn.utils_SchemaGAN import IC_normalization, reverse_IC_normalization
```
Comment (brunozc) [inline CPTSuperLearn/interpolator.py]: isnt it better to just have one utils?

File: CPTSuperLearn/utils_SchemaGAN.py (hunk 5 above)
Comment (EleniSmyrniou) [reply CPTSuperLearn/utils_SchemaGAN.py]: function not used so deleted

File: setup.cfg (hunk 7 above)
Comment (EleniSmyrniou) [reply setup.cfg:26]: In the end pandas are not used

File: requirements.txt (hunk 6 above)
Comment (EleniSmyrniou) [reply requirements.txt:7]: In the end pandas are not used

**Review by brunozc — APPROVED** [review]
